### PR TITLE
Add deprecated process.bindings to node.js 

### PR DESCRIPTION
**Pull request by mohsen1**
https://github.com/nodejs/node/pull/2768

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by ajafff — CHANGES_REQUESTED** [review]
Why add it if it's already deprecated? Without the type declaration you cannot use it anyway.

Also note that the PR you linked is about *process.binding~~s~~*, which is a function.

**Comment by mohsen1** [comment]
@ajafff I've seen many deprecated APIs in this repo. I think we do add them to not break builds

**Review by Flarna — CHANGES_REQUESTED** [review]
There is not `process.bindings`. I can find `process.binding` but this is actually a function taking a `string`

**Comment by mohsen1** [comment]
@Flarna good catch! fixed!

**Comment by mohsen1** [comment]
It's not internal. It's deprecated public api

**Comment by ajafff** [comment]
If it's not documented, it's not intended for public use.

**Comment by mohsen1** [comment]
It's documented in older versions.

File: types/node/index.d.ts [hunk 1]
```diff
@@ -555,6 +555,12 @@ declare namespace NodeJS {
         stderr: WriteStream;
         stdin: ReadStream;
         openStdin(): Socket;
+        /**
+         * @deprecated
+         * Using process.binding() in general should be avoided.
+         * The type checking methods in particular can be replaced by using util.types.
+         */
+        binding: { [key: string]: string; };
```
Comment (Flarna) [inline types/node/index.d.ts]: I think it should be `binding(internalModule: string): any;`.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -555,6 +555,12 @@ declare namespace NodeJS {
         stderr: WriteStream;
         stdin: ReadStream;
         openStdin(): Socket;
+        /**
+         * @deprecated
+         * Using process.binding() in general should be avoided.
+         * The type checking methods in particular can be replaced by using util.types.
```
Comment (Flarna) [inline types/node/index.d.ts]: It's not clear to me what this comment should tell me. What related to `process.binding` should be replaced by `util.types`?

**Review by Flarna — CHANGES_REQUESTED** [review]

**Comment by mohsen1** [comment]
I am closing this because I am convinced that we should not encourage this API usage. Thanks for your help @ajafff @Flarna

**Pull request closed without merging**